pkg/controller/job/job_controller.go

```diff
@@ -796,13 +797,17 @@ func (jm *Controller) getPodsForJob(ctx context.Context, j *batch.Job) ([]*v1.Po
 // syncJob will sync the job with the given key if it has had its expectations fulfilled, meaning
 // it did not expect to see any more of its pods created or deleted. This function is not meant to be invoked
 // concurrently with the same key.
-func (jm *Controller) syncJob(ctx context.Context, key string) (rErr error) {
+func (jm *Controller) syncJob(ctx context.Context, keyUID string) (rErr error) {
```

Comment:
Isn't this always just "key"? Can you provide reference where we actually enqueue keyUID? Curious because I don't see any change when invoking `jm.queue.AddAfter` actually